### DISCUSSION TITLE
Setting custom scopes

### DISCUSSION THREAD

**Discussion by dpetka2001** [cbochs/grapple.nvim#121] [Q&A] [answered]
Just found out this plugin and just experimenting with its options at the moment. I was just playing with the 2 examples provided in the [Scopes API docs](https://github.com/cbochs/grapple.nvim?tab=readme-ov-file#scope-api) and tried the following configuration.
```lua
return {
  {
    "cbochs/grapple.nvim",
    cmd = "Grapple",
    opts = {
      scopes = {
        {
          name = "projects",
          desc = "Project directory",
          -- fallback = "cwd",
          -- cache = { event = "DirChanged" },
          resolver = function()
            local projects_dir = vim.fs.find("projects", {
              upwards = true,
              stop = vim.loop.os_homedir(),
            })

            if #projects_dir == 0 then
              return nil, nil, "Not in projects dir"
            end

            local path = projects_dir[1]
            local id = path
            return id, path, nil
          end,
        },
      },
    },
    -- opts = function()
    --   require("grapple").define_scope({
    --     name = "projects",
    --     desc = "Project directory",
    --     -- fallback = "cwd",
    --     -- cache = { event = "DirChanged" },
    --     resolver = function()
    --       local projects_dir = vim.fs.find("projects", {
    --         upwards = true,
    --         stop = vim.loop.os_homedir(),
    --       })
    --
    --       if #projects_dir == 0 then
    --         return nil, nil, "Not in projects dir"
    --       end
    --
    --       local path = projects_dir[1]
    --       local id = path
    --       return id, path, nil
    --     end,
    --   })
    -- end,
  },
}
```
According to the docs, I should be able to define user's custom scopes during the setup in the `scopes` key, but I couldn't make it work. Only making the `opts` a function and doing `require("grapple").define_scope({})` was I able to make it work. Is this expected behavior?

**Comment by cbochs** [comment, marked as answer]
Hmm... I'm not entirely certain when that broke, but looks like it was a small typo in the Grapple.setup. Should be fixed in c2c6cbf160cbf54e11af29eee3319162910045b6

Thanks!

**Comment by dpetka2001** [comment]
Thank you very much. I have one more question regarding scopes (maybe I missed it in the docs not sure). Are users able to delete the default scopes provided by the plugin?

**Reply by cbochs** [reply]
There is no way (right now) to delete a scope once it has been defined. However, it is something I might consider adding in the future

**Reply by dpetka2001** [reply]
Thanks a lot for your fast replies. Really appreciate the hard work you put into this plugin and hope I may incorporate it in my workflow, because from the first look I really like it. :smile:

**Reply by cbochs** [reply]
@dpetka2001 Just merged #123. You should be able to disable/modify default scopes now, if you would like with:

```lua
-- Disable a default scope
-- Note: be careful to disable default scopes that are used as fallbacks
require("grapple").setup({
    default_scopes = {
        lsp = false
    }
})
```
